### PR TITLE
Feature/45 add find child elements to element

### LINKED ISSUE #45: Add `List<IElement> findChildElements(...)` method to IElement

### PR DESCRIPTION
**Pull request by mialeska**
### PR Details

- Implement FindChildElements functionality for ElementFactory and IElement.
- Reworked locator generation in order to support non-web libraries, such as Appium, where javascript execution for xpath generation is not supported.
- Add missed overloads to findElements.
- Covered all methods by tests.

##### Related Issue Link: 

closes #45 

##### Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/aquality/selenium/core/elements/interfaces/IParent.java [hunk 1]
```diff
@@ -1,8 +1,11 @@
 package aquality.selenium.core.elements.interfaces;
 
 import aquality.selenium.core.elements.ElementState;
+import aquality.selenium.core.elements.ElementsCount;
 import org.openqa.selenium.By;
 
+import java.util.List;
+
 public interface IParent {
```
Comment (DmitryBogatko) [inline src/main/java/aquality/selenium/core/elements/interfaces/IParent.java:9]: IElement extends IParent? (I do not see that Element implements IParent explicitly)

File: src/main/java/aquality/selenium/core/elements/ElementFactory.java [hunk 2]
```diff
@@ -119,14 +135,61 @@ protected void waitForElementsCount(By locator, ElementsCount count, ElementStat
...
+
+    /**
+     * Extracts XPath from passed locator.
+     * Current implementation works only with ByXPath.class and ByTagName locator types,
+     * but you can implement your own for the specific WebDriver type.
+     *
+     * @param locator locator to get xpath from.
+     * @return extracted XPath.
+     */
+    protected String extractXPathFromLocator(By locator) {
```
[the remark this answers is not in the document]
Comment (mialeska) [reply src/main/java/aquality/selenium/core/elements/ElementFactory.java:155]: yes

File: src/main/java/aquality/selenium/core/elements/interfaces/IParent.java (hunk 1 above)
Comment (mialeska) [reply src/main/java/aquality/selenium/core/elements/interfaces/IParent.java:9]: yes it does

File: src/main/java/aquality/selenium/core/elements/Element.java [hunk 3]
```diff
@@ -128,6 +129,16 @@ public void click() {
         return getElementFactory().findChildElement(this, childLoc, name, supplier, state);
     }
 
+    @Override
+    public <T extends IElement> List<T> findChildElements(By childLoc, String name, Class<T> clazz, ElementsCount count, ElementState state) {
```
[the remark this answers is not in the document]
Comment (mialeska) [reply src/main/java/aquality/selenium/core/elements/Element.java]: fixed

**Pull request merged**